fast64_internal/f3d/f3d_parser.py

```diff
@@ -475,6 +475,8 @@ def __init__(self, f3d, basePath, materialContext):
 
         self.tlutAppliedTextures = []  # c name
         self.currentTextureName = None
+        self.imagesLoadedFromFile = [] # image
+        self.CIImageFilesStoredAsFullColor = True # determines whether to apply tlut to file or import as is
```

Comment:
what is `CIImageFilesStoredAsFullColor` for exactly? it's never written to, do you think this should be an option eventually or?

if it should be kept can you make "ci" lowercase so it looks more like a regular field

Reply:
In the future, if support was added for another decomp project that stored CI textures in greyscale, then this could be turned into an option. I would have pushed this dependency up further but I didn't want to modify a bunch of files that might already be in use from the other PRs in the backlog.

Reply:
oh I see. It may be worth expanding the explanation on this then, maybe
```py

        # Determines if images in CI formats loaded from png files,
        # should have the TLUT set by the dlist applied on top of them (False),
        # or if the image file should just be loaded as is with no further change (True)
        # SM64 does (**idk what sm64 does, pls expand**)
        # OoT64 stores CI images as pngs in actual colors (with the TLUT accounted for)
        # For now this can be always True, which covers the (sm64?) and OoT64 use cases
        # In the future this could be an option if for example pngs for CI images were grayscale to represent the palette index.
        self.CIImageFilesStoredAsFullColor = True
```

Reply:
To answer this, SM64 apparently has the luxury of never using CI textures at all.